api/src/main/java/org/openmrs/util/Format.java

```diff
@@ -62,8 +62,25 @@ public static String format(Date date) {
 	public static String format(Date date, FORMAT_TYPE type) {
 		return format(date, Context.getLocale(), type);
 	}
-	
+
+	/**
+	 * This method formats a date object according to a particular locale and returns the date as a string.
+	 * The string can contain only the date (month, day and year), only the time (hours, minutes, seconds) or as a
+	 * timestamp (both date and time).
+	 *
+	 * @param date input date to format as a string
+	 * @param locale input locale to determine how to format the date
+	 * @param type input type to determine how much information from the date is returned
+	 * @return empty string if one of the parameters is null. Otherwise a string object for the date such that it is
+	 * formatted according to locale and the amount of information it contains is determined by type.
+	 * @should not fail with null date
+	 * @should not fail with null locale
```

Comment:
If you want to break the convention, you will need to start a discussion on talk to defend your argument.
We even have a tool that generates a test for each should annotation. https://wiki.openmrs.org/display/docs/Generate+Test+Case+Plugin